src/storage/heap_file.c

```diff
@@ -10806,11 +10807,14 @@ heap_class_get_partition_info (THREAD_ENTRY * thread_p, const OID * class_oid, O
       return ER_FAILED;
     }
 
-  scan_cache.mvcc_snapshot = logtb_get_mvcc_snapshot (thread_p);
-  if (scan_cache.mvcc_snapshot == NULL)
+  if (!tdes->mvccinfo.snapshot.loading)
```

Comment:
1. `is_building" looks more proper, because it is kept "on" during `build_mvcc_info()`
2. I don't think this flag is only for partitioning table. This should be checked and set in the beginning of `build_mvcc_info()` and return it immediately to avoid building mvcc info recursively. If you want to only prevent partitioning case, you should use more specific variable and mechanism.